src/main/java/Main.java

```diff
@@ -8,6 +10,61 @@ public static void main(String[] args) {
     }
 
     public void run(Scanner scanner) {
-        // napisz swój program tutaj. Do wczytywania danych użyj przekazanego w parametrze scannera
+        int number = 0;
+        ArrayList<Integer> numbersList = new ArrayList<>();
+
+        while (!(number < 0)) {
+            System.out.println("Podaj dowolną liczbę całkowitą: ");
+            number = scanner.nextInt();
+            numbersList.add(number);
+        }
+
+        try {
+            numbersList.remove(numbersList.size() - 1);
+
+            Collections.reverse(numbersList);
+
+            for (int i = 0; i < numbersList.size(); i++) {
+                System.out.print(numbersList.get(i));
+                if (!(i + 1 == numbersList.size())) {
+                    System.out.print(", ");
+                }
+            }
+
+            System.out.println();
+
+            Collections.reverse(numbersList);
+
+            StringBuilder sumText = new StringBuilder(String.valueOf(numbersList.get(0)));
+            for (int i = 0; i < numbersList.size(); i++) {
+                if (!(i == 0)) {
+                    sumText.append(" + ").append(numbersList.get(i));
+                }
+            }
+
+            int sum = 0;
+            for (int i : numbersList) {
+                sum = sum + i;
+            }
+
+            System.out.println(sumText + " = " + sum);
+
+            int maximum = numbersList.get(0);
+            int minimum = numbersList.get(0);
+            for (int i = 1; i < numbersList.size(); i++) {
+                if (maximum < numbersList.get(i)) {
+                    maximum = numbersList.get(i);
+                }
+                if (minimum > numbersList.get(i)) {
+                    minimum = numbersList.get(i);
+                }
+            }
+
+            System.out.println("Najmniejsza liczba w liście to " + minimum);
+            System.out.println("Największa liczba w liście to " + maximum);
+
+        } catch (IndexOutOfBoundsException ex) {
```

Comment:
Zamiast przechwytywać wyjątek, lepiej byłoby dodać ifa, który sprawdziłby, czy lista nie jest pusta.
Obsługę wyjątków warto stosować faktycznie w sytuacjach, których nie jesteśmy przewidzieć. Jeżeli natomiast można zastosować prostego ifa, to powinno to być preferowane podejście.